### ISSUE TITLE
IEventPublisher.PublishAsync is not executing synchronously

### ISSUE THREAD

**Issue by devodo** [elsa-workflows/elsa-core#4754]
When calling `IEventPublisher.PublishAsync` I am expecting it to execute synchronously and wait for the triggered workflow to finish before returning.

However after upgrading to the new Elsa `3.0.0` release this is now returning immediately and not blocking. The triggered workflow does execute, however, only after `PublishAsync` returns.  This was working for me as expected on a previous version  `3.0.0-preview.897`

Also note the `ICollection<WorkflowExecutionResult>` returned by `PublishAsync` is empty.

I am publishing an event as follows:

```csharp
// Publish is returning immediately
var result = await eventPublisher.PublishAsync(EventTypes.MyEvent, input: new Dictionary<string, object>() { {"input", input} }, cancellationToken: cancellationToken);
// result is empty
```

The workflow triggered looks like this:

```csharp
public class OnMyEventWorkflow : WorkflowBase
{
    protected override void Build(IWorkflowBuilder builder)
    {
        builder.WorkflowOptions.IncidentStrategyType = typeof(ExceptionMappingIncidentStrategy);
        builder.Version = 1;
        builder.Id = "OnMyEventWorkflow";
        builder.Root = new Sequence
        {
            Activities =
            {
                new Event(EventTypes.MyEvent)
                {
                    CanStartWorkflow = true
                },
                new Inline(async () =>
                {
                    // IEventPublisher.PublishAsync returns before this executes
                    await SomeCallAsync();
                }),
                new Finish()
            }
        };
    }
}
```

**Comment by sfmskywalker** [comment]
I can indeed reproduce this behaviour. Investigating.

**Comment by sfmskywalker** [comment]
I found the issue and will push a fix shortly.

**Comment by sfmskywalker** [comment]
Fixed via 652a617a3b49fef4be63347229cb346b89e84cd8

**Comment by devodo** [comment]
Thanks for the quick fix. This is working for me now.
